### PR TITLE
migration: ui typescript

### PR DESCRIPTION
**Pull request by bang9**
```sh
git rebase --onto main HEAD~16
```

[CLNP-1247](https://sendbird.atlassian.net/browse/CLNP-1247?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ)

Modules

- ./ui/Icon/index.jsx: index.jsx
- ./ui/Label/index.jsx: index.jsx
- ./ui/Label/stringSet.js: stringSet.js
- ./ui/Label/types.js: types.js
- ./ui/Label/utils.js: utils.js
- ./ui/LegacyChannelHeader/index.jsx: index.jsx
- ./ui/LinkLabel/index.jsx: index.jsx
- ./ui/MessageInput/index.jsx: index.jsx
- ./ui/MessageInput/utils.js: utils.js

[CLNP-1247]: https://sendbird.atlassian.net/browse/CLNP-1247?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: types/svg/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+declare module '*.svg' {
+  import * as React from 'react';
+  const ReactComponent: React.FunctionComponent<React.SVGProps<SVGSVGElement> & { title?: string }>;
+  export default ReactComponent;
+}
```
Comment (liamcho) [inline types/svg/index.d.ts:5]: what is this for?

File: src/ui/MessageInput/utils.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+export const sanitizeString = (str?: string) => {
+  return str?.replace(/[\u00A0-\u9999<>]/gim, (i) => ''.concat('&#', String(i.charCodeAt(0)), ';'));
+};
+
+/**
+ * NodeList cannot be used with Array methods
+ * @param {NodeListOf<ChildNode>} childNodes
+ * @returns Array of child nodes
+ */
+export const nodeListToArray = (childNodes?: Node['childNodes'] | null) => {
```
Comment (liamcho) [inline src/ui/MessageInput/utils.ts:13]: why not use type NodeListOf<ChildNode>?

File: src/ui/MessageInput/utils.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  try {
+    return Array.from(childNodes);
+  } catch (error) {
+    return [];
+  }
+};
+
+export function isChannelTypeSupportsMultipleFilesMessage(channel: BaseChannel) {
+  return channel && channel?.isGroupChannel?.() && !channel.isBroadcast && !channel.isSuper;
```
Comment (liamcho) [inline src/ui/MessageInput/utils.ts:22]:
```suggestion
  return channel && channel.isGroupChannel?.() && !channel.isBroadcast && !channel.isSuper;
```

File: src/ui/Label/utils.ts [hunk 4]
```diff
@@ -1,6 +1,7 @@
 import { Typography, Colors } from './types';
+import { ObjectValues } from '../../utils/typeHelpers/objectValues';
 
-export function changeTypographyToClassName(type) {
+export function changeTypographyToClassName(type?: ObjectValues<typeof Typography>) {
```
Comment (liamcho) [inline src/ui/Label/utils.ts:4]: nice

File: src/modules/OpenChannel/components/OpenChannelInput/index.tsx [hunk 5]
```diff
@@ -7,14 +7,8 @@ export type MessageInputWrapperProps = {
...
-const MessageInputWrapper = (props: MessageInputWrapperProps, ref: React.RefObject<HTMLInputElement>): React.ReactNode => {
-  const {
-    currentOpenChannel,
-    disabled,
-    handleSendMessage,
-    handleFileUpload,
-    amIMuted,
-  } = useOpenChannelContext();
+export default React.forwardRef<HTMLInputElement, MessageInputWrapperProps>((props, ref) => {
+  const { currentOpenChannel, disabled, handleSendMessage, handleFileUpload, amIMuted } = useOpenChannelContext();
```
Comment (liamcho) [inline src/modules/OpenChannel/components/OpenChannelInput/index.tsx:11]: multiline is more readable IMO when expanding more than three props.

File: src/modules/Channel/components/MessageInput/useHandleUploadFiles.tsx [hunk 6]
```diff
@@ -40,9 +40,7 @@ export const useHandleUploadFiles = ({
   const uikitMultipleFilesMessageLimit = config?.uikitMultipleFilesMessageLimit;
   const { openModal } = useGlobalModalContext();
 
-  const handleUploadFiles = useCallback((fileList: FileList) => {
-    const files: File[] = Array.from(fileList);
-
+  const handleUploadFiles = useCallback((files: File[]) => {
```
Comment (liamcho) [inline src/modules/Channel/components/MessageInput/useHandleUploadFiles.tsx:43]: nice

File: package.json [hunk 7]
```diff
@@ -71,8 +71,7 @@
     "@sendbird/uikit-tools": "0.0.1-alpha.40",
     "css-vars-ponyfill": "^2.3.2",
     "date-fns": "^2.16.1",
-    "dompurify": "^3.0.1",
-    "prop-types": "^15.7.2"
```
Comment (liamcho) [inline package.json:75]: nice

**Review by liamcho — APPROVED** [review]
LGTM with minor comments

File: types/svg/index.d.ts (hunk 1 above)
Comment (bang9) [reply types/svg/index.d.ts:5]: Imports of files with the .svg extension in Rollup build are bundled together with the code through libraries like svgr. However, in TypeScript, importing files with the .svg extension is not recognized.

To address this issue, the following code has been added to recognize imports of files with the *.svg extension:
```tsx
import SomeSVG from '../assets/some.svg';

const SomeIcon = () => <SomeSvg />
```

**Review by HoonBaek — APPROVED** [review]
Amazing job, thanks!

File: types/svg/index.d.ts (hunk 1 above)
Comment (liamcho) [reply types/svg/index.d.ts:5]: nice

File: src/ui/MessageInput/utils.ts (hunk 2 above)
Comment (bang9) [reply src/ui/MessageInput/utils.ts:13]: I remember there were issues related to TypeScript configuration at the build time, for example, issues like 'set the target to the latest version to use this type'.

**Pull request merged**